pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -241,19 +249,30 @@ func (p *provider) Create(machine *v1alpha1.Machine, _ *cloudprovidertypes.Provi
 			if err != nil {
 				return nil, hzErrorToTerminalError(err, "failed to get network")
 			}
+			if n == nil {
+				return nil, fmt.Errorf("network %q does not exist", network)
+			}
 			serverCreateOpts.Networks = append(serverCreateOpts.Networks, n)
 		}
 	}
 
-	serverCreateOpts.Image, _, err = client.Image.Get(ctx, c.Image)
+	image, _, err := client.Image.Get(ctx, c.Image)
 	if err != nil {
 		return nil, hzErrorToTerminalError(err, "failed to get image")
 	}
+	if image == nil {
+		return nil, fmt.Errorf("image %q does not exist", c.Image)
+	}
+	serverCreateOpts.Image = image
 
-	serverCreateOpts.ServerType, _, err = client.ServerType.Get(ctx, c.ServerType)
+	serverType, _, err := client.ServerType.Get(ctx, c.ServerType)
 	if err != nil {
 		return nil, hzErrorToTerminalError(err, "failed to get server type")
 	}
+	if serverType == nil {
+		return nil, fmt.Errorf("server type %q does not exist", c.ServerType)
+	}
+	serverCreateOpts.ServerType = serverType
```

Comment:
I _hope_ that having nil for Image and ServerType was an error and not intended behaviour for when nothing is configured.

Reply:
Looking at https://github.com/hetznercloud/hcloud-go/blob/master/hcloud/server.go#L286-L295, not having these two fields should definitly blow up.